boot/modeenv.go

```diff
@@ -115,8 +120,18 @@ func (m *Modeenv) Unset() bool {
 	return !m.read
 }
 
-// Write outputs the modeenv to the file at <rootdir>/var/lib/snapd/modeenv.
-func (m *Modeenv) Write(rootdir string) error {
+// Write outputs the modeenv to the file where it was read, only valid on
+// modeenv that has been read
```

Comment:
the comment and the implementation seems out of sync, if it's not read it will be written to the default rootdir. I kind would prefer this to work as per documentation, but maybe there's a reason to keep the convenience?

Reply:
sorry yes the doc got out of sync. The reason I changed it was just aesthetics because I think it looks nicer to be able to say 
```go
m := boot.Modeenv{ ... } 
m.Write() // does the "right" thing
```

and also `m.Write()` looks better to me than `m.WriteTo("")`

but if you prefer I can easily change it such that Write returns error if it was not read somewhere.

Reply:
do we have much code that do that though but tests?  most places that write actually need to read anyway first so they would need. My issue with the current Write is that is mixing very different behavior, also it seems a feature that places that write it from scratch must be deliberate about where it's written.  OTOH I don't care a ton if test code gets a bit uglier as long as the error makes clear what is going wrong.  Regarding failing at runtime, I expect al places writing modeenv to be well tested and to have relatively close read and write.

Reply:
sure I will make it so that `Write()` only works if it was already written. Almost all of these cases are tests anyways, so it's not a big deal to use `WriteTo("")` everywhere in the tests.